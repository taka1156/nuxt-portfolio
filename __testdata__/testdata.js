const MAX_NAV_DATA = 4;
const MAX_PROFILE_DATA = 4;
const MAX_CARD_DATA = 6;
const dummyImgUrl = 'http://placehold.jp/150x150.png';
const GITHUB_URL = 'https://github.com/taka1156/audio-player';
const WEBSITE_URL = 'https://takatechblog.netlify.app/';

// factory
const dummyFactory = (num, fn) => {
  return [...new Array(num)].map((_, i) => fn(i + 1));
};

// template
const dummyNavTemplate = () => ({
  name: 'ダミーナビゲーション',
  to: '/test1',
  img: dummyImgUrl,
});

const dummyCardTemplate = () => ({
  title: 'ダミーカード',
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
});

const dummyCardLinkTemplate = i => ({
  title: 'ダミーカード',
  img: { url: dummyImgUrl },
  content2:
    'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  link: i % 2 === 0 ? GITHUB_URL : WEBSITE_URL,
});

const dummyProfileTemplate = () => {
  return 'ダミープロフ';
};

const dummyImg = {
  imgUrl: dummyImgUrl,
  imgAlt: 'ダミーの画像',
};

const dummyRoutes = dummyFactory(MAX_NAV_DATA, () => dummyNavTemplate());

const dummyCards = dummyFactory(MAX_CARD_DATA, () => dummyCardTemplate());

const dummyCardsLink = dummyFactory(MAX_CARD_DATA, i => dummyCardLinkTemplate(i));

const dummyProfiles = dummyFactory(MAX_PROFILE_DATA, () => dummyProfileTemplate());

const dummyLogo = 'ダミーロゴ';

export {
  dummyImg,
  dummyLogo,
  dummyRoutes,
  dummyCards,
  dummyCardsLink,
  dummyProfiles,
};
