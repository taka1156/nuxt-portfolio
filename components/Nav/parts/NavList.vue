<template>
  <div class="NavList">
    <transition name="nav-fade">
      <div v-show="isOpen" class="nav__list">
        <ul class="nav__items">
          <li
            v-for="(item, index) in routers"
            :key="index"
            class="nav__item"
            @click="changeState"
          >
            <nuxt-link :to="item.to" class="nav__link">
              <img :src="item.icon" class="nav__icon" />
              {{ item.name }}
            </nuxt-link>
          </li>
        </ul>
      </div>
    </transition>
  </div>
</template>

<script>
export default {
  name: 'NavList',
  props: {
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
    routers: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
  methods: {
    changeState() {
      // ナビゲーションバーの開閉管理
      this.$emit('change-state');
    },
  },
};
</script>

<style scoped>
.nav__list {
  display: flex;
  justify-content: center;
  position: fixed;
  top: 60px;
  right: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 2;
  background-color: rgba(12, 12, 12, 0.9);
}

.nav__items {
  list-style-type: none;
  padding: 0;
}

.nav__item {
  display: block;
  font-size: 50px;
  text-align: left;
  color: white;
}

.nav__link {
  display: block;
  color: white;
  text-decoration: none;
}

.nav__icon {
  height: 50px;
  width: 50px;
}

/* nav-contentsのアニメーション */
.nav-fade-enter,
.nav-fade-leave-to {
  opacity: 0;
}

.nav-fade-enter-to,
.nav-fade-leave {
  opacity: 1;
}

.nav-fade-enter-active,
.nav-fade-leave-active {
  transition: opacity 0.6s ease;
}
</style>
