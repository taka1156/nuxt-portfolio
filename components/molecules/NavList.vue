<template>
  <div>
    <transition name="nav-fade">
      <div v-show="isOpen" class="nav__list">
        <ul>
          <nav-item
            v-for="(item, index) in routers"
            :key="index"
            class="nav__item"
            :nav-item="item"
            @change-state="$emit('change-state')"
          />
        </ul>
      </div>
    </transition>
  </div>
</template>

<script>
import NavListItem from './NavListItem';

export default {
  name: 'NavList',
  components: {
    'nav-item': NavListItem,
  },
  props: {
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
    routers: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
};
</script>

<style scoped>
/* css reset */
ul {
  margin: 0px;
  padding: 0px;
}
/* css reset */

.nav__list {
  display: flex;
  justify-content: center;
  position: fixed;
  top: 60px;
  right: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 2;
  background-color: rgba(12, 12, 12, 0.9);
}

/* nav-contentsのアニメーション */
.nav-fade-enter,
.nav-fade-leave-to {
  opacity: 0;
}

.nav-fade-enter-to,
.nav-fade-leave {
  opacity: 1;
}

.nav-fade-enter-active,
.nav-fade-leave-active {
  transition: opacity 0.6s ease;
}
</style>
