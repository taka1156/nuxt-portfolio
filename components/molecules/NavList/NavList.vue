<template>
  <div>
    <transition name="nav-fade">
      <div v-show="isOpen" class="nav-list">
        <ul>
          <li
            v-for="(item, index) in routes"
            :key="`nav_${index}`"
            @click="changeState"
          >
            <nav-list-item :nav-item="item" @change-state="changeState" />
          </li>
        </ul>
      </div>
    </transition>
  </div>
</template>

<script>
import NavListItem from '../NavListItem/NavListItem';

export default {
  name: 'NavList',
  components: {
    'nav-list-item': NavListItem
  },
  props: {
    /**
     * ナビゲーションの開閉状態
     */
    isOpen: {
      type: Boolean,
      required: true
    },
    /**
     * ナビゲーションの項目が定義された配列
     */
    routes: {
      type: Array,
      required: true
    }
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (閉じる)
       * @event changeState
       */
      this.$emit('change-state', false);
    }
  }
};
</script>

<style scoped>
/* css reset */
ul,
li {
  list-style-type: none;
}

/* css reset */
.nav-list {
  position: fixed;
  top: 60px;
  right: 0;
  left: 0;
  z-index: 3;
  display: flex;
  justify-content: center;
  width: 100vw;
  height: 100vh;
  background-color: rgba(12, 12, 12, 0.9);
}

/* nav-contentsのアニメーション */
.nav-fade-enter,
.nav-fade-leave-to {
  opacity: 0;
}

.nav-fade-enter-to,
.nav-fade-leave {
  opacity: 1;
}

.nav-fade-enter-active,
.nav-fade-leave-active {
  transition: opacity 0.6s ease;
}
</style>
