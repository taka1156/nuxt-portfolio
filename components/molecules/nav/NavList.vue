<template>
  <div>
    <transition name="nav-fade">
      <div v-show="isOpen" class="nav-list">
        <ul>
          <nav-item
            v-for="(item, index) in routes"
            :key="index"
            :nav-item="item"
            @change-state="changeState"
          />
        </ul>
      </div>
    </transition>
  </div>
</template>

<script>
import NavListItem from './NavListItem';
export default {
  name: 'NavList',
  components: {
    'nav-item': NavListItem,
  },
  props: {
    /**
     * ナビゲーションの開閉状態
     */
    isOpen: {
      type: Boolean,
      default: false,
      required: true,
    },
    /**
     * ナビゲーションの項目が定義された配列
     */
    routes: {
      type: Array,
      default: () => [],
      required: true,
    },
  },
  methods: {
    changeState() {
      /**
       * ナビゲーション開閉の状態を変化させる
       * (閉じる)
       * @event changeState
       */
      this.$emit('change-state', false);
    },
  },
};
</script>

<style scoped>
.nav-list {
  display: flex;
  justify-content: center;
  position: fixed;
  top: 60px;
  right: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 2;
  background-color: rgba(40, 167, 69, 0.9);
}

/* nav-contentsのアニメーション */
.nav-fade-enter,
.nav-fade-leave-to {
  opacity: 0;
}
.nav-fade-enter-to,
.nav-fade-leave {
  opacity: 1;
}
.nav-fade-enter-active,
.nav-fade-leave-active {
  transition: opacity 0.6s ease;
}
</style>
