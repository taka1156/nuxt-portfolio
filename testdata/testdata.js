const dummyLogo = 'ダミー';

const dummyRoutes = [
  {
    name: 'nav-item',
    to: '/test1',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test2',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test3',
    img: 'http://placehold.jp/150x150.png',
  },
  {
    name: 'nav-item',
    to: '/test4',
    img: 'http://placehold.jp/150x150.png',
  },
];

const dummyProfile = [
  'TypeScriptの学習',
  'Laravel+Vueでサービスを作る',
  'AWSなどインフラ周りの学習',
  'CSS/SASSの学習',
];

const dummyCards = [
  {
    title: 'PHP',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '1~4ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Laravel',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '2~3ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'JavaScript(ES6)',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '６ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
  {
    title: 'Vue.js',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: '8ヶ月',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
  },
];

const dummyCardsLink = [
  {
    title: 'ブログサイト',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Nuxt.js(Vue.js), axios, microCMS, BootStrap4, vue-infinite-loading',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://takatechblog.netlify.app/',
  },
  {
    title: 'マークダウンエディター',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, BootStrap4',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/markdown-editor',
  },
  {
    title: 'LineBot',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Express.js, axios, @line/bot-sdk , Vercel(Zeit Now)',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/line-bot',
  },
  {
    title: 'オーディオプレイヤー',
    img: { url: 'http://placehold.jp/150x150.png' },
    content1: 'Vue.js, Electron, JSMediaTags',
    content2:
      'この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れてい',
    link: 'https://github.com/taka1156/audio-player',
  },
];

export { dummyLogo, dummyRoutes, dummyProfile, dummyCards, dummyCardsLink };
