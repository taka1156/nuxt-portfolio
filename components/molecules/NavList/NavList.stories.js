import NavList from './NavList.vue';
import { dummyRoutes } from '@/__testdata__/testdata.js';

export default {
  title: 'Molecules/NavList',
  component: NavList,
  argTypes: {
    isOpen: {
      control: {
        type: 'boolean'
      }
    }
  }
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { NavList },
  template: '<nav-list v-bind="$props" />'
});

export const Default = Template.bind({});
Default.args = { isOpen: false, routes: dummyRoutes };
