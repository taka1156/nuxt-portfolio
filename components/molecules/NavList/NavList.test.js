import { mount, RouterLinkStub } from '@vue/test-utils';
import NavList from './NavList.vue';
import { dummyRoutes } from '@/__testdata__/testdata.js';

describe('NavList', () => {
  const navList = propsData =>
    mount(NavList, {
      stubs: {
        NuxtLink: RouterLinkStub
      },
      propsData: {
        routes: dummyRoutes,
        ...propsData
      }
    });

  it('NavList初期値: isOpen, routes', () => {
    const wrapper = navList({ isOpen: false });
    // isOpen
    expect(wrapper.vm.$options.props.isOpen.required).toBe(true);
    expect(wrapper.vm.isOpen).toBe(false);
    // routes
    expect(wrapper.vm.$options.props.routes.required).toBe(true);
    expect(wrapper.vm.routes).toBe(dummyRoutes);
  });

  it('値がDOMに反映されているか', () => {
    const wrapper = navList({ isOpen: false });
    const aTag = wrapper.findComponent(RouterLinkStub);
    const imgTag = wrapper.find('img');

    expect(aTag.props().to).toBe(dummyRoutes[0].to);
    expect(aTag.text()).toBe(dummyRoutes[0].name);
    expect(imgTag.attributes().src).toBe(dummyRoutes[0].img);
  });

  it('isOpenがtrueの時に`.nav-list`が表示', () => {
    const wrapper = navList({ isOpen: true });
    expect(wrapper.find('.nav-list').element.style.display).not.toBe('none');
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('isOpenがfalseの時に`.nav-list`が非表示', () => {
    const wrapper = navList({ isOpen: false });
    expect(wrapper.find('.nav-list').element.style.display).toBe('none');
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('changeStateを実行した時、`change-state`のイベントが発生するか', () => {
    const wrapper = navList({ isOpen: true });
    wrapper.vm.changeState();
    expect(wrapper.emitted('change-state')).toBeTruthy();
  });
});
