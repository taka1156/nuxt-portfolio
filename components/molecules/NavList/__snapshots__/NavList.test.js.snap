// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList isOpenがfalseの時に\`.nav-list\`が非表示 1`] = `
"<div>
  <transition-stub name=\\"nav-fade\\">
    <div class=\\"nav-list\\" style=\\"display: none;\\">
      <ul>
        <li>
          <div class=\\"nav-item\\"><a class=\\"nav-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\">
              <p class=\\"base-text--extend\\" use-type=\\"nav\\">ダミーナビゲーション</p>
            </a></div>
        </li>
        <li>
          <div class=\\"nav-item\\"><a class=\\"nav-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\">
              <p class=\\"base-text--extend\\" use-type=\\"nav\\">ダミーナビゲーション</p>
            </a></div>
        </li>
        <li>
          <div class=\\"nav-item\\"><a class=\\"nav-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\">
              <p class=\\"base-text--extend\\" use-type=\\"nav\\">ダミーナビゲーション</p>
            </a></div>
        </li>
        <li>
          <div class=\\"nav-item\\"><a class=\\"nav-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\">
              <p class=\\"base-text--extend\\" use-type=\\"nav\\">ダミーナビゲーション</p>
            </a></div>
        </li>
      </ul>
    </div>
  </transition-stub>
</div>"
`;

exports[`NavList isOpenがtrueの時に\`.nav-list\`が表示 1`] = `
"<div>
  <transition-stub name=\\"nav-fade\\">
    <div class=\\"nav-list\\">
      <ul>
        <li>
          <div class=\\"nav-item\\"><a class=\\"nav-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\">
              <p class=\\"base-text--extend\\" use-type=\\"nav\\">ダミーナビゲーション</p>
            </a></div>
        </li>
        <li>
          <div class=\\"nav-item\\"><a class=\\"nav-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\">
              <p class=\\"base-text--extend\\" use-type=\\"nav\\">ダミーナビゲーション</p>
            </a></div>
        </li>
        <li>
          <div class=\\"nav-item\\"><a class=\\"nav-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\">
              <p class=\\"base-text--extend\\" use-type=\\"nav\\">ダミーナビゲーション</p>
            </a></div>
        </li>
        <li>
          <div class=\\"nav-item\\"><a class=\\"nav-item__link\\"><img src=\\"http://placehold.jp/150x150.png\\" alt=\\"ダミーナビゲーションの画像\\" class=\\"base-img base-img--lg\\">
              <p class=\\"base-text--extend\\" use-type=\\"nav\\">ダミーナビゲーション</p>
            </a></div>
        </li>
      </ul>
    </div>
  </transition-stub>
</div>"
`;
